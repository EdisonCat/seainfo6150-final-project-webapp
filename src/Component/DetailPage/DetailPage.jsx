import React, { useEffect, useState } from "react";
import styles from "./DetailPage.module.css";
import { useParams, useHistory } from "react-router-dom";

export default function DetailPage() {
	const [name, setName] = useState("");
	const [description, setDescription] = useState("");
	const [thumbnail, setThumbnail] = useState("");
	const [comics, setComics] = useState([]);
	const { id } = useParams();
	const history = useHistory();
	useEffect(() => {
		const fetchData = async (id) => {
			const url = `${process.env.REACT_APP_MARVEL_CHARACTERS_URL}${id}${process.env.REACT_APP_MARVEL_API_HASH}`;
			const res = await fetch(url, {
				method: "GET",
				headers: { "Content-Type": "application/json" },
			});
			const json = await res.json();
			if (!json.data) {
				history.push({ pathname: "/notfound" });
				return;
			}
			const character = json.data.results[0];
			setName(character.name);
			setDescription(character.description);
			setComics(character.comics.items);
			setThumbnail(
				`${character.thumbnail.path}.${character.thumbnail.extension}`
			);
		};
		fetchData(id);
	}, [id, history]);
	const handleComics = (idx) => {
		const arr = comics[idx].resourceURI.split("/");
		const comicsId = arr[arr.length - 1];
		history.push({
			pathname: `/comics/${comicsId}`,
			state: { url: comics[idx].resourceURI },
		});
	};

	return (
		<div className={styles.container}>
			<div className={styles.thumbnail}>
				<img alt="thumbnail" src={thumbnail} />
			</div>
			<div className={styles.info}>
				<div>
					<h1>{name}</h1>
					<br />
					<h4>Description: {description}</h4>
				</div>
				<h4>Comics: </h4>
				<div className={styles.comics}>
					{comics.map((comic, idx) => (
						<div
							className={styles.link}
							key={idx}
							onClick={() => handleComics(idx)}
						>
							{comic.name}
						</div>
					))}
				</div>
			</div>
		</div>
	);
}