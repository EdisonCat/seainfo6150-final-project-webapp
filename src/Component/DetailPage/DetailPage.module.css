@import url('https://fonts.googleapis.com/css2?family=Work+Sans:wght@300&display=swap');
.container {
    display: flex;
    background: #333;
    color: white;
    border-top: 1px solid gray;
    padding: 50px 20px 50px 20px;
    min-height: 800px;
	font-family: 'Work Sans', sans-serif;
}
.thumbnail {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-right: 50px;
}
.thumbnail img {
    width: 100%;
    height: auto;
}
.comics {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    gap: 20px
}
.link {
    color: white;
    cursor: pointer;
    text-decoration: underline;
    text-decoration-color: #333;
    transition: 0.5s;
}
.link:hover {
    text-decoration-color: red;
}
.info {
    display: flex;
    flex-direction: column;
}
img {
    width: 100%;
    height: auto;
}
@media screen and (max-width: 1200px){
    .comics {
        grid-template-columns: repeat(3, 1fr);
    }
}
@media screen and (max-width: 800px){
    .comics {
        grid-template-columns: repeat(2, 1fr);
    }
    .container {
        flex-direction: column;
        align-items: center;
    }
    .thumbnail {
        margin-right: 0;
    }
}
@media screen and (max-width: 600px){
    .comics {
        grid-template-columns: 1fr;
    }
}