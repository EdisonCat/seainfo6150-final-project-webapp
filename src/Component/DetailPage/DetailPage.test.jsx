import React from "react";
import DetailPage from "./DetailPage.jsx";
it("rendered DetailPage correctly", () => {
	// eslint-disable-next-line no-undef
    const { container } = render(<DetailPage />);
    expect(container).toMatchSnapshot();
});