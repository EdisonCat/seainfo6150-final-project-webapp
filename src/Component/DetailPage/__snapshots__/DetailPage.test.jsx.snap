// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered DetailPage correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="thumbnail"
    >
      <img
        alt="thumbnail"
        src=""
      />
    </div>
    <div
      class="info"
    >
      <div>
        <h1 />
        <br />
        <h4>
          Description: 
          
        </h4>
      </div>
      <h4>
        Comics: 
      </h4>
      <div
        class="comics"
      />
    </div>
  </div>
</div>
`;
